kingpin/actors/aws/base.py

```diff
@@ -83,19 +85,34 @@ def __init__(self, *args, **kwargs):
 
         super(AWSBaseActor, self).__init__(*args, **kwargs)
 
-        if not (aws_settings.AWS_ACCESS_KEY_ID and
+        # By default, we will try to let Boto handle discovering its
+        # credentials at instantiation time. This _can_ result in synchronous
+        # API calls to the Metadata service, but those should be fast.
+        key = None
+        secret = None
+
+        # In the event though that someone has explicitly set the AWS access
+        # keys in the environment (either for the purposes of a unit test, or
+        # because they wanted to), we use those values.
+        if (aws_settings.AWS_ACCESS_KEY_ID and
```

Comment:
That makes sense, but if no variables are passed in then maybe tests can
populate the expected environment variables? Not sure about the best
practice here

On Sun, Apr 17, 2016, 11:13 AM Matt W <notifications@github.com> wrote:

> In kingpin/actors/aws/base.py
> <https://github.com/Nextdoor/kingpin/pull/332#discussion_r59989514>:
>
> > @@ -83,19 +85,34 @@ def __init__(self, *args, **kwargs):
> >
> >          super(AWSBaseActor, self).__init__(*args, **kwargs)
> >
> > -        if not (aws_settings.AWS_ACCESS_KEY_ID and
> > +        # By default, we will try to let Boto handle discovering its
> > +        # credentials at instantiation time. This _can_ result in synchronous
> > +        # API calls to the Metadata service, but those should be fast.
> > +        key = None
> > +        secret = None
> > +
> > +        # In the event though that someone has explicitly set the AWS access
> > +        # keys in the environment (either for the purposes of a unit test, or
> > +        # because they wanted to), we use those values.
> > +        if (aws_settings.AWS_ACCESS_KEY_ID and
>
> The only reason I put in the option for overriding it was for the unit
> tests -- its easier to do the forcing of the settings in the aws.settings
> module and then reloading that, than actually mocking everything out. Maybe
> there is a more clever way to globally mock out the aws settings so that we
> know for sure that during a unit test we're not reaching out to an API.
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/Nextdoor/kingpin/pull/332/files/a48953254dee245c46ee3f3b517d7332bfb69cb3#r59989514>
>